ts/indicators/supertrend.src.ts

```diff
@@ -0,0 +1,651 @@
+/* *
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import H from '../parts/Globals.js';
+
+/**
+ * Internal types
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        interface SupertrendLinkedParentPointObject extends Point {
+            close: number;
+            index: number;
+            x: number;
+        }
+
+        interface SupertrendLinkedParentObject extends Series {
+            data: Array<SupertrendLinkedParentPointObject>;
+            points: Array<SupertrendLinkedParentPointObject>;
+            xData: Array<number>;
+            yData: Array<Array<number>>;
+        }
+
+        interface SupertrendGappedExtensionObject {
+            options?: SupertrendGappedExtensionOptions;
+        }
+
+        interface SupertrendGappedExtensionOptions {
+            gapSize?: number;
+        }
+
+        interface SupertrendGroupedPointsObject {
+            bottom: Array<Partial<Point>>;
+            intersect: Array<Partial<Point>>;
+            top: Array<Partial<Point>>;
+        }
+
+        interface SupertrendLineOptionsObject {
+            [index: string]: (Dictionary<CSSObject>|undefined);
+        }
```

Comment:
As far as I see, this interface is not part of the API options and therefore should be called something ending on `Object`, but not `OptionsObject` or `Options`. Additionally I am not sure, if this interface has any benefit, as it is only used once and can be replaced by `Dictionary<(Dictionary<CSSObject>|undefined)>`.

Reply:
Ok, will fix the naming :+1: 

Regarding the interface I'm too was unsure, but I felt that putting it in a interface is slightly clearer as (at least to me) `Dictionary<(Dictionary<CSSObject>|undefined)>` is somewhat harder to grasp.